packages/graphql/tests/integration/directives/auth/roles.int.test.ts

```diff
@@ -866,4 +867,91 @@ describe("auth/roles", () => {
             }
         });
     });
+
+    describe("combining roles with where", () => {
+        test("combines where with roles", async () => {
+            const session = driver.session();
+
+            const type = generateUniqueType("User");
+
+            const typeDefs = `
+                type ${type.name} {
+                    id: ID
+                    name: String
+                    password: String
+                }
+
+                extend type ${type.name}
+                    @auth(
+                        rules: [
+                            {
+                                roles: ["user"]
+                                where: { id: "$jwt.id" }
+                            }
+                            {
+                                roles: ["admin"]
+                            }
+                        ]
+                    )
+            `;
+
+            const userId = generate({
+                charset: "alphabetic",
+            });
+
+            const userId2 = generate({
+                charset: "alphabetic",
+            });
+
+            const query = `
+                query {
+                    ${type.plural} {
+                        id
+                        name
+                        password
+                    }
+                }
+            `;
+
+            const neoSchema = new Neo4jGraphQL({ typeDefs, config: { jwt: { secret } } });
+
+            try {
+                await session.run(`
+                    CREATE (:${type.name} {id: "${userId}", name: "User1", password: "password" })
+                    CREATE (:${type.name} {id: "${userId2}", name: "User2", password: "password" })
+                `);
+                // request with role "user" - should only return details of user
+                const userReq = createJwtRequest(secret, { roles: ["user"], id: userId });
+
+                const gqlResultUser = await graphql({
+                    schema: neoSchema.schema,
+                    source: query,
+                    contextValue: { driver, req: userReq, driverConfig: { bookmarks: session.lastBookmark() } },
+                });
+
+                expect(gqlResultUser.data).toEqual({
+                    [type.plural]: [{ id: userId, name: "User1", password: "password" }],
+                });
+
+                // request with role "admin" - should return all users
```

Comment:
As we are trying to test 2 separate flows (`request with role "user"` and `request with role "admin"`) How about splitting this test case into 2 different test cases (using these comments as the test title) and using the top `describe` block for the setup?

Reply:
Given that this is more of an opinionated comment compared to the Markdown TCK test and ESLint violations, how about we leave this one be in favour of getting this PR merged?

Reply:
I don't think this is a big deal, so I'm happy to merge as is, but I wouldn't say that this being 2 separate test cases, by definition, is opinionated

Reply:
In my opinion, it could be easily seen as one test case, because it is showing that users with different roles can query the same instance of the schema with the same data and receive the expected results.

Reply:
I'm going to merge because there is bigger fish to fry and this has been open for **4 months**. 😨